### PR TITLE
DCOS-47014: Select input component

### PR DESCRIPTION
**Pull request by mperrotti**
Added a component for a `<select>` input

screen shot 2019-01-23 at 5 49 43 pm

### CODE REVIEW

File: packages/selectInput/README.md [hunk 1]
```diff
@@ -0,0 +1,3 @@
+# SelectInput Name
```
Comment (mperrotti) [inline packages/selectInput/README.md]: @lsamimi - I couldn't find any existing documentation on select inputs. Maybe we should collaborate on some?

File: packages/selectInput/components/SelectInput.tsx [hunk 2]
```diff
@@ -0,0 +1,181 @@
...
+            </span>
+            {getHintContent}
+            {appearance !== SelectInputAppearance.Error && getValidationErrors}
+          </div>
+        )}
+      </FormFieldWrapper>
+    );
+  }
+
+  private handleFocus(e) {
```
Comment (mperrotti) [inline packages/selectInput/components/SelectInput.tsx]: I really don't like that this component has to have state just for focus styles...

I may look into detecting `:focus-within` support, and if the browser doesn't have it, don't set `outline: 0` on the input. I won't lose any sleep if somebody using Edge/IE doesn't get a pretty focus state :x

**Comment by mperrotti** [comment]
There's some work here that could be used in `TextInput`:
- Use `FormFieldWrapper` component instead of making the component handle everything
- Share`formStyles.ts` styles between `SelectInput` component _and_ `TextInput`

I'd like the opinion of reviewers: should I add that work in this PR, or open a new one on the next spike day? Or do you disagree with sharing this code altogether?

**Comment by GeorgiSTodorov** [comment]
Doing these two tasks in this PR, or as a follow-up PR in this sprint sounds like a good idea to me.

**Comment by mperrotti** [comment]
Hold merge until I get the TextInput changes in too (see this comment: https://github.com/dcos-labs/ui-kit/pull/283#issuecomment-457016322)

File: packages/selectInput/components/SelectInput.tsx [hunk 3]
```diff
@@ -0,0 +1,181 @@
...
+  padding,
+  visuallyHidden,
+  display
+} from "../../shared/styles/styleUtils";
+import Icon from "../../icon/components/Icon";
+import { SystemIcons } from "../../icons/dist/system-icons-enum";
+import { iconSizeXs } from "../../design-tokens/build/js/designTokens";
+import FormFieldWrapper from "../../shared/components/FormFieldWrapper";
+
+export enum SelectInputAppearance {
```
Comment (natmegs) [inline packages/selectInput/components/SelectInput.tsx]: Can we also share this with `TextInput`?

Comment (mperrotti) [reply packages/selectInput/components/SelectInput.tsx]: Good idea. I just have to figure out the right place to put it 🤔

**Comment by GeorgiSTodorov** [comment]
File conflicts.

File: packages/selectInput/components/SelectInput.tsx [hunk 4]
```diff
@@ -0,0 +1,195 @@
...
+        {({ getValidationErrors, isValid, getHintContent, describedByIds }) => (
+          <div>
+            <label
+              className={cx(getLabelStyle(hasError), {
+                [visuallyHidden]: !showInputLabel
+              })}
+              htmlFor={id}
+            >
+              {inputLabel}
+              {this.props.required ? (
```
Comment (nLight) [inline packages/selectInput/components/SelectInput.tsx]: we could add `required` to the destructing `const {} = this.props`;

File: packages/selectInput/README.md (hunk 1 above)
Comment (lsamimi) [reply packages/selectInput/README.md]: @mperrotti yes, I'll set up a meeting with you and we can write some doc on this.

File: packages/selectInput/components/SelectInput.tsx [hunk 5]
```diff
@@ -0,0 +1,195 @@
...
+    if (this.props.disabled) {
+      return "disabled";
+    }
+    if (this.state.hasFocus) {
+      return `${this.props.appearance}-focus`;
+    }
+    return this.props.appearance;
+  }
+
+  // private getDescribedBy(hintContent, errors) {
```
[the remark this answers is not in the document]
Comment (mperrotti) [reply packages/selectInput/components/SelectInput.tsx]: 😅 how embarassing

**Comment by mperrotti** [comment]
@GeorgiSTodorov - I'm going based off of the design team's mocks for rendering hint and error text. This is also consistent with how we do it for the `TextInput`, so I think we should keep it as-is

**Pull request merged**